### DISCUSSION TITLE
Trouble using develop version to try indexed full text search

### DISCUSSION THREAD

**Discussion by TwiggyWan** [zim-desktop-wiki/zim-desktop-wiki#2662] [Development]
Hey this is my first time running zim from the repo.
I have trouble running Find within my code blocks.

problem 1: 
I try to test #2489 

I try to test it (`./zim.py` from latest develop commit) but get `sqlite version 3.43.0 or higher - Failed`.

I run zim.py from a python 3.12 virtualenv in which i have installed sqlite 3.46.0 and confirmed it was installed ok with  `python3 -c 'import sqlite3; print(sqlite3.sqlite_version_info)' `

Should I install sqlite3 differently?

I went through PLUGIN_WRITING.md already but did not find an answer.

**Comment by TwiggyWan** [comment]
I went a bit farther about #64 : 

find.py searches in what is given by TextBuffer.py, but this replaces blocks of stuff : 
```
{{{code: lang="sh" linenumbers="False"
echo 'XXXX
}}}

```
by the appropriate number of line breaks.

good starting point for the debugger for me is in find.py:_check_range() and insert_object_model_at_cursor().

i dont get yet if we have access to the data within each {{{ }}} block.

**Comment by jaap-karssenberg** [comment]
I'm have a private branch where I'm working on enabling find in embedded
objects. So please have some patience for that to be resolved.

Op ma 26 aug. 2024 20:14 schreef TwiggyWan ***@***.***>:

**Comment by jaap-karssenberg** [comment]
Or do you mean _search_ in embedded objects?  For search I don't know how
the SQLite plugin handles that now. But would be easy to use text
representation of objects in the indexing.

Op di 27 aug. 2024 10:21 schreef Jaap Karssenberg <
***@***.***>:

**Reply by maclomhair** [reply]
I think the indexed search plugin currently indexes all text tokens of a page concatenated together, and I think, embedded objects are indexed as well implicitly...

AFAIU objects are stored in the element tree as elements with their text representation added in the data field. When requesting to iterate through the element tree, text tokens are then created for all elements which have data added to them. So this would mean that the indexed search plugin should already index the text representation of embedded objects as well.

**Comment by TwiggyWan** [comment]
> I'm have a private branch where I'm working on enabling find in embedded
>objects.

Awesome. looking forward to reading more about this!

will keep the discussion open for now as I still struggle to make the sqlite plugin work..

**Reply by maclomhair** [reply]
So did you manage to get the sqlite plugin working? From what I saw in your original post, I think it should have worked - after all, the indexed search plugin also checks the version literally by calling `sqlite3.sqlite_version_info`

**Reply by TwiggyWan** [reply]
hey, not yet. still struggling with debian as there is a version mis match. (trying to get help over their forums: https://forums.debian.net/viewtopic.php?t=160285 )

**Comment by maclomhair** [comment]
Ok I think I have an idea of what might be going on: maybe, by calling `./zim.py` like that, zim uses the system-wide interpreter (defined in line 1 of `zim.py`) and doesn't respect locally installed packages?

Could you try instead to call zim by executing `python3 zim.py` in your virtual environment and see if that works?
